vmc/resource_vmc_site_recovery.go

```diff
@@ -0,0 +1,183 @@
+/* Copyright 2020 VMware, Inc.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package vmc
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/validation"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/lib/vapi/std/errors"
+	"github.com/vmware/vsphere-automation-sdk-go/services/vmc/draas"
+	"github.com/vmware/vsphere-automation-sdk-go/services/vmc/draas/model"
+)
+
+func resourceSiteRecovery() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSiteRecoveryCreate,
+		Read:   resourceSiteRecoveryRead,
+		Delete: resourceSiteRecoveryDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"sddc_id": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "SDDC identifier",
+			},
+			"srm_extension_key_suffix": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 13),
+				Description:  "Custom extension key suffix for SRM. If not specified, default extension key will be used. The custom extension suffix must contain 13 characters or less, be composed of letters, numbers, ., - characters only. The extension suffix must begin and end with a letter or number. The suffix is appended to com.vmware.vcDr- to form the full extension key",
+			},
+			"site_recovery_state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Site recovery state. Possible values are: ACTIVATED, ACTIVATING, CANCELED, DEACTIVATED, DEACTIVATING, DELETED, FAILED",
```

Comment:
possible values state values can be stored in a string[] and validated here

Reply:
This is a computed value. They are returned as a part of the response on GET. The possible values are only being mentioned in the description for reference.